homeassistant/components/adguard/__init__.py

```diff
@@ -115,14 +115,15 @@ async def refresh(call: ServiceCall) -> None:
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload AdGuard Home config entry."""
-    hass.services.async_remove(DOMAIN, SERVICE_ADD_URL)
-    hass.services.async_remove(DOMAIN, SERVICE_REMOVE_URL)
-    hass.services.async_remove(DOMAIN, SERVICE_ENABLE_URL)
-    hass.services.async_remove(DOMAIN, SERVICE_DISABLE_URL)
-    hass.services.async_remove(DOMAIN, SERVICE_REFRESH)
-
     unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
     if unload_ok:
+        hass.data[DOMAIN].pop(entry.entry_id)
+    if not hass.data[DOMAIN]:
+        hass.services.async_remove(DOMAIN, SERVICE_ADD_URL)
```

Comment:
We normally count the loaded config entries before removing a service.

Example:
https://github.com/home-assistant/core/blob/110d9232cd527b471f883e67bfc8e3db0163775c/homeassistant/components/simplisafe/__init__.py#L428-L437

Reply:
This does effectively the same, as each entry for this integration is registered in this namespace.

Reply:
We should unregister the service when the last loaded config entry is unloaded. That's good.

Reply:
The benefit of using the core interface to check for loaded config entries is that we don't depend on integration details that may change.